From 4248a1ab470470277f5853d290ccd795f3a6e8fd Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Tue, 22 Jan 2013 09:53:10 +0100
Subject: [PATCH 25/53] apps: fix format string warnings due to size_t usage
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix all warnings of type: warning: format ‘%d’ expects argument of type
‘int’, but argument 3 has type ‘size_t’ [-Wformat].

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
---
 src/apps/bindx_test.c  | 8 ++++----
 src/apps/peel_client.c | 2 +-
 src/apps/peel_server.c | 2 +-
 src/apps/sctp_darn.c   | 2 +-
 src/apps/sctp_test.c   | 5 ++---
 5 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/src/apps/bindx_test.c b/src/apps/bindx_test.c
index ec7632e..b1c9ffa 100644
--- a/src/apps/bindx_test.c
+++ b/src/apps/bindx_test.c
@@ -66,13 +66,13 @@ int main (int argc, char **argv)
 	cnt = 2;    /* argument iterator */
 	buf_ptr = addr_buf;
 	while (cnt < argc) {
-		printf("bindx_test: INFO: Arg %d: %s", cnt, argv[cnt]);
+		printf("bindx_test: INFO: Arg %zu: %s", cnt, argv[cnt]);
 		fflush(stderr);
 		if (strchr(argv[cnt], ':')) {
 			struct sockaddr_in6 *sa6; 
 
 			sa6 = (struct sockaddr_in6 *)buf_ptr;
-			printf(" IPv6 address number %d", addrs);
+			printf(" IPv6 address number %zu", addrs);
 			sa6->sin6_family = AF_INET6;
 			sa6->sin6_port = port;
 			if (inet_pton(AF_INET6, argv[cnt], &sa6->sin6_addr)) {
@@ -85,7 +85,7 @@ int main (int argc, char **argv)
 			struct sockaddr_in *sa; 
 
 			sa = (struct sockaddr_in *)buf_ptr;
-			printf (" IPv4 address number %d", addrs);
+			printf (" IPv4 address number %zu", addrs);
 			sa->sin_family = AF_INET;
 			sa->sin_port = port;
 			if (inet_pton (AF_INET, argv[cnt], &sa->sin_addr)) {
@@ -100,7 +100,7 @@ int main (int argc, char **argv)
 		cnt++;
 	}
 
-	printf ("bindx_test: INFO: Got %d addrs\n", addrs);
+	printf ("bindx_test: INFO: Got %zu addrs\n", addrs);
   
 	/* Create the socket */
 	sd = socket(PF_INET6, SOCK_SEQPACKET, IPPROTO_SCTP);
diff --git a/src/apps/peel_client.c b/src/apps/peel_client.c
index ede3fbc..2c3f234 100644
--- a/src/apps/peel_client.c
+++ b/src/apps/peel_client.c
@@ -453,7 +453,7 @@ my_sctpReadInput(int fd)
 	errno = 0;
 	sz = recvmsg(fd,&msg,0);
 	if(sz <= 0){
-		printf("Read returns %d errno:%d control len is %d msgflg:%x\n",
+		printf("Read returns %d errno:%d control len is %zu msgflg:%x\n",
 		       sz,errno,
 		       msg.msg_controllen,msg.msg_flags);
 	}
diff --git a/src/apps/peel_server.c b/src/apps/peel_server.c
index 524b471..9b9b065 100644
--- a/src/apps/peel_server.c
+++ b/src/apps/peel_server.c
@@ -436,7 +436,7 @@ my_sctpReadInput(int fd,int maxread)
 	msg.msg_controllen = sizeof(controlVector);
 	errno = 0;
 	sz = recvmsg(fd,&msg,0);
-	printf("Read fd:%d returns %d errno:%d control len is %d msgflg:%x\n",
+	printf("Read fd:%d returns %d errno:%d control len is %zu msgflg:%x\n",
 	       fd,
 	       sz,errno,
 	       msg.msg_controllen,
diff --git a/src/apps/sctp_darn.c b/src/apps/sctp_darn.c
index c326319..9843a8a 100644
--- a/src/apps/sctp_darn.c
+++ b/src/apps/sctp_darn.c
@@ -1378,7 +1378,7 @@ user_test_check_message(struct msghdr *msg,
 
 	if (msg->msg_controllen != controllen) {
 		fprintf(stderr,
-			"Got control structure of length %d, not %d\n",
+			"Got control structure of length %zu, not %d\n",
 			msg->msg_controllen, controllen);
 		exit(1);
 	}
diff --git a/src/apps/sctp_test.c b/src/apps/sctp_test.c
index 7287d05..0e4695b 100644
--- a/src/apps/sctp_test.c
+++ b/src/apps/sctp_test.c
@@ -341,9 +341,8 @@ print_message(const int sk, struct msghdr *msg, size_t msg_len) {
         if (!(MSG_NOTIFICATION & msg->msg_flags)) {
                 int index = 0;
 
-		DEBUG_PRINT(DEBUG_MIN, "Data %d bytes.", msg_len);
-
-		DEBUG_PRINT(DEBUG_MAX, " First %d bytes: ",
+		DEBUG_PRINT(DEBUG_MIN, "Data %zu bytes.", msg_len);
+		DEBUG_PRINT(DEBUG_MAX, " First %zu bytes: ",
 				    (msg_len < BODYSIZE)?msg_len:BODYSIZE);
                 /* Make sure that everything is printable and that we
                  * are NUL terminated...
-- 
1.7.11.7

